rpaas/acl.py

```diff
@@ -0,0 +1,162 @@
+        self.acl_api_host = config.get("ACL_API_HOST")
+        self.acl_api_user = config.get("ACL_API_USER")
+        self.acl_api_password = config.get("ACL_API_PASSWORD")
+        self.acl_api_timeout = int(config.get("ACL_API_TIMEOUT", 30))
+        self.acl_port_range_start = config.get("ACL_PORT_RANGE_START", "30000")
+        self.acl_port_range_end = config.get("ACL_PORT_RANGE_END", "61000")
+        self.network_api_url = config.get("NETWORK_API_URL", None)
+        self.acl_auth_basic = HTTPBasicAuth(self.acl_api_user, self.acl_api_password)
+        self.lock_manager = lock_manager
+        self.lock_name = "acl_manager:{}".format(self.service_name)
+        if self.network_api_url:
+            self.network_api_username = config.get("NETWORK_API_USERNAME")
+            self.network_api_password = config.get("NETWORK_API_PASSWORD")
+            self.ip_client = Ip.Ip(self.network_api_url, self.network_api_username, self.network_api_password)
+            self.network_client = Network.Network(self.network_api_url, self.network_api_username,
+                                                  self.network_api_password)
+
+    def add_acl(self, name, src, dst):
+        src_network = self._get_network_from_ip(src)
+        if src_network == src:
+            src_network = str(ipaddress.ip_network(unicode(src_network)))
+        src = str(ipaddress.ip_network(unicode(src)))
+        dst = self._get_network_from_ip(dst)
+        if self._check_acl_exists(name, src, dst):
+            return
+        request_data = self._request_data("permit", name, src, dst)
+        instance_lock = "{}:{}".format(self.lock_name, name)
+        if self.lock_manager.lock(instance_lock, timeout=(self.acl_api_timeout * 2)):
+            try:
+                response = self._make_request("PUT", "api/ipv4/acl/{}".format(src_network), request_data)
+                self._check_acl_response(response)
+                self.storage.store_acl_network(name, src, dst)
+            finally:
+                self.lock_manager.unlock(instance_lock)
+        else:
+            raise AclApiError("could not get lock for {} instance".format(name))
+
+    def remove_acl(self, name, src):
+        src = str(ipaddress.ip_network(unicode(src)))
+        acls = self.storage.find_acl_network(name, src)
+        if not acls:
+            return
+        destinations = [dst for acl in acls if src == acl['source'] for dst in acl['destination']]
+        for dst in destinations:
+            request_data = self._request_data("permit", name, src, dst, True)
+            for env, vlan, acl_id in self._iter_on_acl_query_results(request_data):
+                instance_lock = "{}:{}".format(self.lock_name, name)
+                if self.lock_manager.lock(instance_lock, timeout=(self.acl_api_timeout * 2)):
+                    try:
+                        response = self._make_request("DELETE", "api/ipv4/acl/{}/{}/{}".format(env, vlan, acl_id), None)
```

Comment:
Is there a way to identify not found responses here (is it a 404 response)? It might make sense to ignore not found responses to avoid errors due to inconsistencies caused failed partial deletes or manually deleted entries.